

#' A table of potential adducts.
#' 
#' The data is based on the Metabolomics FieHn Lab's Mass Spectrometry Adduct
#' Calculator. It provides the basis for calculating the m/z of the ion forms
#' of known metabolites.
#' 
#' 
#' @name adduct.table
#' @docType data
#' @format A data frame with 47 observations on the following 4 variables.
#' \describe{ \item{adduct}{The ion form.} \item{divider}{The value to divide
#' the neutral mass by.} \item{addition}{The value to add after dividing.}
#' \item{charge}{The charge state of the ion form.} }
#' @references Huang N.; Siegel M.M.1; Kruppa G.H.; Laukien F.H. Automation of
#' a Fourier transform ion cyclotron resonance mass spectrometer for
#' acquisition, analysis, and e-mailing of high-resolution exact-mass
#' electrospray ionization mass spectral data. J Am Soc Mass Spectrom 1999, 10,
#' 1166â€“1173.
#' @source
#' http://fiehnlab.ucdavis.edu/staff/kind/Metabolomics/MS-Adduct-Calculator/
#' @keywords datasets
#' @examples
#' 
#' data(metabolite.table)
#' data(adduct.table)
#' known.table.example<-make.known.table(metabolite.table[1001:1020,], adduct.table[1:4,])
#' 
NULL





#' sample data after alignment, processed by the machine learning approach
#' 
#' This is the aligned feature data from 4 sample LC/MS profiles.
#' 
#' 
#' @name aligned.learn
#' @docType data
#' @format The format is: List of 4 $ aligned.ftrs: A matrix with m/z, median
#' elution time, and signal strength in each profile.  $ pk.times : A matrix
#' with m/z, median elution time, and elution time in each profile.  $ mz.tol :
#' the m/z tolerance level in alignment $ chr.tol : the retention time
#' tolerance level in alignment
#' @source Dean Jones lab. Emory University, School of Medicine.
#' @keywords datasets
#' @examples
#' 
#' data(aligned)
#' 
NULL





#' sample data after alignment
#' 
#' This is the aligned feature data from 4 sample LC/MS profiles.
#' 
#' 
#' @name aligned
#' @docType data
#' @format The format is: List of 4 $ aligned.ftrs: A matrix with m/z, median
#' elution time, and signal strength in each profile.  $ pk.times : A matrix
#' with m/z, median elution time, and elution time in each profile.  $ mz.tol :
#' the m/z tolerance level in alignment $ chr.tol : the retention time
#' tolerance level in alignment
#' @source Dean Jones lab. Emory University, School of Medicine.
#' @keywords datasets
#' @examples
#' 
#' data(aligned)
#' 
NULL





#' Adaptive processing of LC/MS data
#' 
#' The package generates a feature table from a batch of LC/MS spectra. It
#' finds m/z and retention time tolerance levels from the data. A run-filter is
#' used to detect peaks and remove noise. Non-parametric statistical methods
#' are used to find-tune peak selection and grouping. After retention time
#' correction, a feature table is generated by aligning peaks across spectra.
#' 
#' 
#' @name apLCMS-package
#' @aliases apLCMS-package apLCMS
#' @docType package
#' @author Tianwei Yu <tyu8@@emory.edu> Maintainer: Tianwei Yu
#' <tyu8@@emory.edu>
#' @seealso
#' @references Bioinformatics. 25(15):1930-36.  BMC Bioinformatics. 11:559.
#' @keywords package
#' @examples
#' 
#' 
NULL





#' Adaptive processing of LC/MS data
#' 
#' The package generates a feature table from a batch of LC/MS spectra. It
#' finds m/z and retention time tolerance levels from the data. A run-filter is
#' used to detect peaks and remove noise. Non-parametric statistical methods
#' are used to find-tune peak selection and grouping. After retention time
#' correction, a feature table is generated by aligning peaks across spectra.
#' 
#' 
#' @name apLCMS-package
#' @aliases apLCMS-package apLCMS
#' @docType package
#' @author Tianwei Yu <tyu8@@emory.edu> Maintainer: Tianwei Yu
#' <tyu8@@emory.edu>
#' @seealso
#' @references Bioinformatics. 25(15):1930-36.  BMC Bioinformatics. 11:559. J.
#' Proteome Res. 12(3):1419-27.
#' @keywords package
#' @examples
#' 
#' 
NULL





#' Sample feature tables from 4 profiles. The original feature detection is
#' done by machine learning approach.
#' 
#' A list object containing 4 matrices, each of which is the feature table from
#' a profile.
#' 
#' 
#' @name features.learn
#' @docType data
#' @format List object containing multiple matrices. One matrix from each
#' spectrum.
#' @references
#' @source Data from Dean Jones lab, Emory University School of Medicine.
#' @keywords datasets
#' @examples
#' 
#' data(features.learn)
#' 
NULL





#' Sample feature tables from 4 profiles
#' 
#' A list object containing 4 matrices, each of which is the feature table from
#' a profile.
#' 
#' 
#' @name features
#' @docType data
#' @format List object containing multiple matrices. One matrix from each
#' spectrum.
#' @references
#' @source Data from Dean Jones lab, Emory University School of Medicine.
#' @keywords datasets
#' @examples
#' 
#' data(features)
#' 
NULL





#' Feature tables after elution time correction. The original feature detection
#' is done by machine learning approach.
#' 
#' A list object containing 4 matrices, each of which is from a profile. The
#' elution times in the matrices are corrected.
#' 
#' 
#' @name features2.learn
#' @docType data
#' @format A list object. Each component of the list is a matrix.
#' @references
#' @source Raw data from Dean Jones lab, Emory School of Medicine.
#' @keywords datasets
#' @examples
#' 
#' data(features2.learn)
#' 
NULL





#' Feature tables after elution time correction.
#' 
#' A list object containing 4 matrices, each of which is from a profile. The
#' elution times in the matrices are corrected.
#' 
#' 
#' @name features2
#' @docType data
#' @format A list object. Each component of the list is a matrix.
#' @references
#' @source Raw data from Dean Jones lab, Emory School of Medicine.
#' @keywords datasets
#' @examples
#' 
#' data(features2)
#' 
NULL





#' A known feature table based on HMDB.
#' 
#' This table was compiled from HMDB metabolites. It contains only the H+, Na+,
#' K+ and NH4+ derivative of known metabolites. It suites the need for
#' analyzing LC/MS data generated with anion exchange chromatography.
#' 
#' 
#' @name known.table.common.pos
#' @docType data
#' @format A data frame containing the known metabolite ions and previously
#' found features. It contains 18 columns: "chemical_formula": the chemical
#' formula if knonw; "HMDB_ID": HMDB ID if known; "KEGG_compound_ID": KEGG
#' compound ID if known; "neutral.mass": the neutral mass if known: "ion.type":
#' the ion form, such as H+, Na+, ..., if known; "m.z": m/z value, either
#' theoretical for known metabolites, or mean observed value for unknown but
#' previously found features; "Number_profiles_processed": the total number of
#' LC/MS profiles that were used to build this database; "Percent_found": in
#' what percentage was this feature found historically amount all data
#' processed in building this database; "mz_min": the minimum m/z value
#' observed for this feature; "mz_max": the maximum m/z value observed for this
#' feature; "RT_mean": the mean retention time observed for this feature;
#' "RT_sd": the standard deviation of retention time observed for this feature;
#' "RT_min": the minimum retention time observed for this feature; "RT_max":
#' the maximum retention time observed for this feature; "int_mean.log.": the
#' mean log intensity observed for this feature; "int_sd.log.": the standard
#' deviation of log intensity observed for this feature; "int_min.log.": the
#' minimum log intensity observed for this feature; "int_max.log.": the maximum
#' log intensity observed for this feature;
#' @source Wishart, D. S., et al. (2009). HMDB: a knowledgebase for the human
#' metabolome. Nucleic Acids Res 37, D603-10.
#' @keywords datasets
#' @examples
#' 
#' data(known.table.common.pos)
#' 
NULL





#' A known feature table based on HMDB.
#' 
#' This table was compiled from HMDB metabolites. It contains only the H+
#' derivative of known metabolites. It suites the need for analyzing LC/MS data
#' generated with anion exchange chromatography.
#' 
#' 
#' @name known.table.hplus
#' @docType data
#' @format A data frame containing the known metabolite ions and previously
#' found features. It contains 18 columns: "chemical_formula": the chemical
#' formula if knonw; "HMDB_ID": HMDB ID if known; "KEGG_compound_ID": KEGG
#' compound ID if known; "neutral.mass": the neutral mass if known: "ion.type":
#' the ion form, such as H+, Na+, ..., if known; "m.z": m/z value, either
#' theoretical for known metabolites, or mean observed value for unknown but
#' previously found features; "Number_profiles_processed": the total number of
#' LC/MS profiles that were used to build this database; "Percent_found": in
#' what percentage was this feature found historically amount all data
#' processed in building this database; "mz_min": the minimum m/z value
#' observed for this feature; "mz_max": the maximum m/z value observed for this
#' feature; "RT_mean": the mean retention time observed for this feature;
#' "RT_sd": the standard deviation of retention time observed for this feature;
#' "RT_min": the minimum retention time observed for this feature; "RT_max":
#' the maximum retention time observed for this feature; "int_mean.log.": the
#' mean log intensity observed for this feature; "int_sd.log.": the standard
#' deviation of log intensity observed for this feature; "int_min.log.": the
#' minimum log intensity observed for this feature; "int_max.log.": the maximum
#' log intensity observed for this feature;
#' @source Wishart, D. S., et al. (2009). HMDB: a knowledgebase for the human
#' metabolome. Nucleic Acids Res 37, D603-10.
#' @keywords datasets
#' @examples
#' 
#' data(known.table.hplus)
#' 
NULL





#' A known metabolite table based on HMDB.
#' 
#' This table was compiled from HMDB metabolites. It contains only the basic
#' information of known metabolites. It can be used to produce feature tables
#' with ion forms of the users' choice.
#' 
#' It is to be used in combination with the object "adduct.table", to create
#' feature table with ion forms of the user's choice. Which ion form to choose
#' should be based on the LC/MS system.
#' 
#' @name metabolite.table
#' @docType data
#' @format A data frame containing the known metabolites. It contains 4
#' columns: "chemical_formula": the chemical formula of the known table;
#' "HMDB_ID": HMDB ID; "KEGG_compound_ID": KEGG compound ID if known; "mass":
#' the neutral mass;
#' @source Wishart, D. S., et al. (2009). HMDB: a knowledgebase for the human
#' metabolome. Nucleic Acids Res 37, D603-10.
#' @keywords datasets
#' @examples
#' 
#' data(metabolite.table)
#' data(adduct.table)
#' known.table.example<-make.known.table(metabolite.table[1001:1020,], adduct.table[1:4,])
#' 
NULL





#' Feature data after alignment and weak signal recovery. The initial peak
#' detection is done by machine learning approach.
#' 
#' This is the aligned feature data from 4 sample profiles after weaker signal
#' recovery.
#' 
#' 
#' @name new.aligned.learn
#' @docType data
#' @format The format is: List of 4 $ aligned.ftrs: A matrix with m/z, median
#' elution time, and signal strength in each spectrum.  $ pk.times : A matrix
#' with m/z, median elution time, and elution time in each spectrum.  $ mz.tol
#' : num 1.14e-05 (the m/z tolerance level in alignment) $ chr.tol : num 79.21
#' (the elution time tolerance level in alignment)
#' @references
#' @source Dean Jones lab. Emory University, School of Medicine.
#' @keywords datasets
#' @examples
#' 
#' data(new.aligned.learn)
#' 
NULL





#' Feature data after alignment and weak signal recovery
#' 
#' This is the aligned feature data from 4 sample profiles after weaker signal
#' recovery.
#' 
#' 
#' @name new.aligned
#' @docType data
#' @format The format is: List of 4 $ aligned.ftrs: A matrix with m/z, median
#' elution time, and signal strength in each spectrum.  $ pk.times : A matrix
#' with m/z, median elution time, and elution time in each spectrum.  $ mz.tol
#' : num 1.14e-05 (the m/z tolerance level in alignment) $ chr.tol : num 79.21
#' (the elution time tolerance level in alignment)
#' @references
#' @source Dean Jones lab. Emory University, School of Medicine.
#' @keywords datasets
#' @examples
#' 
#' data(new.aligned)
#' 
NULL



